pkg/overlay/config.go

```diff
@@ -32,6 +32,11 @@ type Config struct {
 	RefreshInterval time.Duration `help:"the interval at which the cache refreshes itself in seconds" default:"30s"`
 }
 
+type CtxKey int
```

Comment:
Passing non-constant values via context is frowned upon. It's intended for request scoped values. See:
1. https://golang.org/pkg/context/#WithValue
2. https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39

Reply:
Okay so Dylan and I talked about this for a while (before this PR) and decided that for now this is still the right approach even given the frowning.

Here's the goal - The PointerDB provider.Responsibility needs access to the Overlay Responsibility, just like the Overlay Responsibility needs access to the Kademlia Responsibility. Responsibilities are currently constructed via a Config struct that has a Run method, so there isn't a great way currently to have Responsibilities depend on other Responsibilities directly.

Here are the options:
1) Have PointerDB use gRPC loopback to talk to the Overlay (clean, but senseless performance loss)
2) Redesign our config, provider, and responsibility system to support providing instantiated dependencies directly. It's not clear how this would turn out and could result in a lot of code churn. There are potentially wide-ranging non-local effects that will change our config and process system.
3) Pass the overlay to the pointerdb via the provider context, just like kademlia is passed to the overlay. Not quite a compiler error if we get the order wrong, but the process will fail to start.

I think option 3 is the least code, is performant, and is pragmatic. There is immediate feedback if the system is ever used wrong and the context abuse is a very limited scope. 

The upsides outweigh the downsides in my opinion.

Reply:
Also, usually when there is a diversion from common practice, it's useful to describe the reasoning in the comments to make it clear what the reasons were that lead to such situation. It helps to people new to the code or people seeing the code to orient to the problem and trade-offs.